### ISSUE TITLE
[Feature request] Deployments to ASG Warm pool instances using CodeDeploy lifecycle hooks

### ISSUE THREAD

**Issue by pawelhajski** [aws/aws-codedeploy-agent#390]
Currently, CodeDeploy lifecycle hooks in EC2 Auto Scaling are not working with Warm pool instances in Warmed:Pending:Wait state. Please make it possible for CodeDeploy to work with Warm pool instances.

**Comment by cocest** [comment]
I am currently experiencing the exact problem, and this happens you have a warm pool attached to your AWS auto scaling group (ASG). I am using AWS launch template's user data. Even though my cloud configuration script invoke AWS lifecycle hook to complete the process. If I am to SSH into the launch instance I can complete the lifecycle hook manually. My only conclusion is that the lifecycle hook is only invoked on `Warmed:Pending:Wait` and never `Pending:Wait` event for it to transition into `inService` state.

One of the way to overcome this issue is to deamonise your bash script to complete the lifecycle hook on system startup.

**Comment by slmdeepu** [comment]
This is very important and helpful feature. Please make it possible for CodeDeploy to work with Warm pool instances.

**Comment by james-astra** [comment]
Yes it is a shame that this does not seem to be supported out of the box. Given the tight integrations between ASG & CodeDeploy, it is very surprising it is not already supported.

**Comment by joelgaria** [comment]
Any news about this request?
